src/processes.c

```diff
@@ -2083,20 +2087,24 @@ static int ps_read (void)
 			case 'T': stopped++;  break;
 			case 'W': paging++;   break;
 		}
+		if (ps_get_cmdline (
+		     pid, NULL, cmdline, sizeof(cmdline)) != NULL) {
+			cmdline_processes++;
+		}
 
-		ps_list_add (ps.name,
-				ps_get_cmdline (pid, ps.name, cmdline, sizeof (cmdline)),
-				&pse);
+		ps_list_add (ps.name, cmdline, &pse);
```

Comment:
I'd restore the previous code here -- after the latest change, `cmdline` is potentially empty, which this code tried to avoid (by defaulting to `ps.name`).

Reply:
Done. Not the same call but similar semantics.

Reply:
Not quite. You *are* changing the semantics here -- the old code would have used "`[foo]`" for a process named "foo", and your new code will produce just "`foo`". Frankly, I see no harm in just calling `ps_get_cmdline` twice.

Reply:
PTAL.